SYCL/ESIMD/api/functional/functions/functions_atomic_update_inc_dec.cpp

```diff
@@ -0,0 +1,230 @@
+//==------- functions_atomic_update_inc_dec.cpp  - DPC++ ESIMD on-device
+//          test -----------------------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu, level_zero
+// XREQUIRES: gpu
+// TODO gpu and level_zero in REQUIRES due to only this platforms supported yet.
+// The current "REQUIRES" should be replaced with "gpu" only as mentioned in
+// "XREQUIRES".
+// UNSUPPORTED: cuda, hip
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Test for simd atomic_update function.
+// The test uses increment and decrement atomic operators. Invokes atomic_update
+// with different offset and simd_mask values.
+// It is expected that values that should be changed has been changed.
+
+#include "../sycl_nd_range.hpp"
+#include "common.hpp"
+#include "functions_atomic_update.hpp"
+
+using namespace esimd_test::api::functional;
+using namespace sycl::ext::intel::esimd;
+
+// Descriptor class for the case of calling bitwise not operator.
+struct atomic_update_0_operands {
+  static std::string get_description() {
+    return "atomic update with zero operands";
+  }
+
+  template <typename DataT, int NumElems, int NumElemsToChange,
+            typename ChangeElemFilterT, atomic_op ChosenOperator>
+  static void simd_function(DataT *input_data, DataT *output_data,
```

Comment:
```suggestion
  static void esimd_function(DataT *input_data, DataT *output_data,
```
or
```suggestion
  static void call_esimd_function(DataT *input_data, DataT *output_data,
```
to avoid misinterpretation as reference to the `simd<T,N>` class

Reply:
I've applied this comment in this commit: https://github.com/intel/llvm-test-suite/pull/939/commits/8f6071023226997f11c6ff346027fdfcf14b5c24